Cabal/Distribution/Backpack/ModSubst.hs

```diff
@@ -0,0 +1,53 @@
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE PatternGuards #-}
+
+-- | A type class 'ModSubst' for objects which can have 'ModuleSubst'
+-- applied to them.
+
+module Distribution.Backpack.ModSubst (
```

Comment:
In the newest patchset, the IndefModule/IndefUnitId that ModSubst acts on are defined in Distribution.Backpack. Do you want to swing this type class into that module? I don't see that much benefit.